### PR TITLE
Integrate machine edits with the Studio, apply received edits in the VS Code

### PR DESCRIPTION
**Pull request by Andarist**
closes https://github.com/statelyai/xstate-tools/issues/163
closes https://github.com/statelyai/xstate-tools/issues/66
closes https://github.com/statelyai/xstate-tools/issues/104
closes https://github.com/statelyai/xstate-tools/issues/134
closes https://github.com/statelyai/xstate-tools/issues/211
closes https://github.com/statelyai/xstate-tools/issues/223
closes https://github.com/statelyai/xstate-tools/issues/228
closes https://github.com/statelyai/xstate-tools/issues/238
closes https://github.com/statelyai/xstate-tools/issues/239

### CODE REVIEW

File: apps/extension/server/src/server.ts [hunk 1]
```diff
@@ -565,12 +559,39 @@ connection.onRequest('getMachineAtCursorPosition', ({ uri, position }) => {
   };
 });
 
+connection.onRequest('applyMachineEdits', ({ machineEdits }) => {
+  if (!displayedMachine) {
+    throw new Error(
+      '`applyMachineEdits` can only be requested when there is a displayed machine',
```
Comment (davidkpiano) [inline apps/extension/server/src/server.ts:565]: Where does this error get displayed?

File: apps/extension/server/src/server.ts [hunk 2]
```diff
@@ -565,12 +559,39 @@ connection.onRequest('getMachineAtCursorPosition', ({ uri, position }) => {
...
+    );
+  }
+
+  const modified = documentsCache
+    .get(displayedMachine.uri)!
```
Comment (davidkpiano) [inline apps/extension/server/src/server.ts:570]: Nit: can we guarantee this always exists?

File: apps/extension/server/src/server.ts (hunk 1 above)
Comment (Andarist) [reply apps/extension/server/src/server.ts:565]: Excellent question - it won't be displayed anywhere. This error has just been added to satisfy TS as it should be almost impossible to end up with it. I don't like "almost" in this sentence though and gonna try to look into providing stronger guarantees around this.

File: apps/extension/server/src/server.ts (hunk 2 above)
Comment (Andarist) [reply apps/extension/server/src/server.ts:570]: Yes, we can't have a displayed machine without having a cached document for it.

**Review by mellson — APPROVED** [review]

**Pull request merged**